x-pack/platform/plugins/shared/lens/public/app_plugin/shared/edit_on_the_fly/helpers.ts

```diff
@@ -125,7 +135,16 @@ export const getSuggestions = async (
     };
 
     const allSuggestions =
-      suggestionsApi({ context, dataView, datasourceMap, visualizationMap }) ?? [];
+      suggestionsApi({
+        context,
+        dataView,
+        datasourceMap,
+        visualizationMap,
+        preferredChartType,
+        preferredVisAttributes: preferredVisAttributes
+          ? injectESQLQueryIntoLensLayers(preferredVisAttributes, query)
+          : undefined,
```

Comment:
I'm playing around this PR with the following query:
```
FROM kibana_sample_data_flights | KEEP Carrier, AvgTicketPrice, DestCountry | LIMIT 10 | SORT DestCountry
```
Adding or removing the SORT at the end (that doesn't change the table columns, but just its sorting) can change/switch my chart configuration. In particular I've noticed that I can't keep my a stacked bar chart horizontally rotated if I add or remove that SORT at the end.

Reply:
Marco it works for me:

![meow](https://github.com/user-attachments/assets/aba2fe58-8751-47dc-b844-0e87a3b7790b)

Reply:
Great catch Marco, it is because the shortcut was registered on the editor init so the submit method didnt have the most recent attributes https://github.com/elastic/kibana/pull/210780/commits/5208427a02d317b62198db59b6ec14911ccbcb1b

(I also memoized the editor options but is irrelevant with this fix, I just saw that we do an extra rerender)